test/refract/test-JsonSchema.cc

```diff
@@ -0,0 +1,706 @@
+            }
+        }
+    }
+
+    GIVEN("An empty ArrayElement with fixed attribute true")
+    {
+        auto el = make_empty<ArrayElement>();
+        el->attributes().set("fixed", from_primitive(true));
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches an empty array")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","type":"array","enum":[[]]})");
+            }
+        }
+    }
+
+    GIVEN("An empty ArrayElement with fixedType attribute true")
+    {
+        auto el = make_empty<ArrayElement>();
+        el->attributes().set("fixedType", from_primitive(true));
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches an empty array")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","type":"array","enum":[[]]})");
+            }
+        }
+    }
+
+    GIVEN("An ArrayElement with empty string as content")
+    {
+        auto el = make_element<ArrayElement>(make_empty<StringElement>());
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches any array")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","type":"array"})");
+            }
+        }
+    }
+
+    GIVEN("An ArrayElement with empty string as content and fixed attribute true")
+    {
+        auto el = make_element<ArrayElement>(make_empty<StringElement>());
+        el->attributes().set("fixed", from_primitive(true));
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches arrays of a single entry that is a string")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","type":"array","items":[{"type":"string"}]})");
+            }
+        }
+    }
+
+    GIVEN("A ArrayElement with empty string as content and fixedType attribute true")
+    {
+        auto el = make_element<ArrayElement>(make_empty<StringElement>());
+        el->attributes().set("fixedType", from_primitive(true));
+
+        WHEN("a JSON Schema is generated from it")
+        {
+            auto result = schema::generateJsonSchema(*el);
+
+            THEN("the schema matches arrays of strings")
+            {
+                REQUIRE(to_string(result) == R"({"$schema":"http://json-schema.org/draft-04/schema#","type":"array","items":{"type":"string"}})");
+            }
+        }
+    }
+
+    GIVEN("An ArrayElement with fixed string as content")
```

Comment:
Fixed string?

Reply:
A string with fixed attribute true. Should I be more verbose?

Reply:
I'm not sure what do you mean by "fixed string".
It seems like "normal" StringElement. I do not see fixed there.
e.g. for NumberElement with content you do not use "fixed number"

Maybe i just wrong understand to meaning

Reply:
No, you're right. I mistakenly associated your comment with the expressions on 346-347.